src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Services
+{
+    /// <summary>
+    /// A directory structure representing the paths of an OpenAPI document.
+    /// </summary>
+    public class OpenApiUrlTreeNode
+    {
+        /// <summary>
+        /// All the subdirectories of a node.
+        /// </summary>
+        public IDictionary<string, OpenApiUrlTreeNode> Children { get; } = new Dictionary<string, OpenApiUrlTreeNode>();
+
+        /// <summary>
+        /// The relative directory path of the current node from the root node.
+        /// </summary>
+        public string Path { get; private set; } = "";
+
+        /// <summary>
+        /// Dictionary of labels and Path Item objects that describe the operations available on a node.
+        /// </summary>
+        public IDictionary<string, OpenApiPathItem> PathItems { get; } = new Dictionary<string, OpenApiPathItem>();
```

Comment:
in [our version](https://github.com/microsoft/kiota/blob/b8c8c87e71d8a9db38b2a5bfc20d894350b748d5/src/Kiota.Builder/OpenApiUriSpaceNode.cs#L16) path item is singular (not a dictionary). I'm not asking to change anything, I'm just curious about the motivation for the change.

Reply:
Having it as a dictionary of Labels and PathItem objects will enable us to easily represent the different OpenAPI operations exposed by a single path in e.g. both V1.0 and Beta versions of the Graph metadata. In this case, the versions (labels) will be dictionary keys.

Reply:
Thanks for providing more details. I'm not sure I follow entirely here. As beta and v1.0 are base paths for the API shouldn't we end up in a situation where we either have:
- a gigantic open api description with all the paths, everything starting with beta or v1
- two open api descriptions, one for beta, one for v1, with a basepath

In which case, why do we need that labeling capability?

Reply:
Both scenarios are feasible with this new structure. 

**Scenario 1**: If for example you start with translating `v1.0` OpenAPI doc. into an `OpenApiUrlTreeNode` object, you can later attach the `beta` document onto it. If a path in the `beta` version has already been defined in the `v1.0` version, no new path will be created, but rather its operations (`OpenApiPathItem`) will be appended to the existing `v1.0` path in the node, and this will be labelled as `beta` (for example) within the `PathItems` dictionary. Agora makes use of this scenario.

**Scenario 2**: If you want to have separate `OpenApiUrlTreeNode` objects, one representing `v1.0 `and the other `beta`, that's perfectly doable. In each case, all the segments for a given `OpenApiUrlTreeNode` object  will have their `PathItems` dictionary key as the specified version passed in the `Create(OpenApiDocument doc, string label)` method.

Reply:
perfect! thanks for going through all the details with me!